domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +192,310 @@ class TopicListController @Inject constructor(
       .build()
   }
 
-  private fun createOngoingStoryListFromProgress(
+  private fun createUpcomingTopicSummaryFromJson(
+    topicId: String,
+    jsonObject: JSONObject
+  ): UpcomingTopic {
+    val upcomingTopic = UpcomingTopic.newBuilder()
```

Comment:
Just removing this and in line `214`, writing this directly as 
```
    return UpcomingTopic.newBuilder()
      .setTopicId(topicId)
      .setName(jsonObject.getString("topic_name"))
      .setVersion(jsonObject.optInt("version"))
      .setTopicPlayAvailability(topicPlayAvailability)
      .setLessonThumbnail(createTopicThumbnail(jsonObject))
      .build()
```

Reply:
Done.